nibabel/brikhead.py

```diff
@@ -0,0 +1,487 @@
+        if floatfacs is None or not np.any(floatfacs):
+            return None
+        scale = np.ones(self.info['DATASET_RANK'][1])
+        floatfacs = np.atleast_1d(floatfacs)
+        scale[floatfacs.nonzero()] = floatfacs[floatfacs.nonzero()]
+
+        return scale
+
+    def get_slope_inter(self):
+        """Use `self.get_data_scaling()` instead"""
+        return None, None
+
+    def get_data_offset(self):
+        """Data offset in BRIK file"""
+        return DATA_OFFSET
+
+    def get_volume_labels(self):
+        """Returns volume labels
+
+        Returns
+        -------
+        labels : list of str
+
+        Examples
+        --------
+        >>> header = AFNIHeader(parse_AFNI_header('example4d+orig.HEAD'))
+        >>> header.get_volume_labels()
+        ['#0', '#1', '#2']
+        """
+        labels = self.info.get('BRICK_LABS', None)
+        if labels is not None:
+            labels = labels.split('~')
+
+        return labels
+
+
+class AFNIImage(SpatialImage):
+    """AFNI Image file
+
+    Can be loaded from either the BRIK or HEAD file (but MUST specify one!)
+
+    Examples
+    --------
+    >>> brik = load('example4d+orig.BRIK.gz')
+    >>> brik.shape
+    (33, 41, 25, 3)
+    >>> brik.affine
+    array([[ -3.    ,  -0.    ,  -0.    ,  49.5   ],
+           [ -0.    ,  -3.    ,  -0.    ,  82.312 ],
+           [  0.    ,   0.    ,   3.    , -52.3511],
+           [  0.    ,   0.    ,   0.    ,   1.    ]])
+    >>> head = load('example4d+orig.HEAD')
+    >>> np.array_equal(head.get_data(), brik.get_data())
+    True
+    """
+
+    header_class = AFNIHeader
+    valid_exts = ('.brik', '.head')
+    files_types = (('image', '.brik'), ('header', '.head'))
+    _compressed_suffixes = ('.gz', '.bz2')
+    makeable = False
+    rw = False
+    ImageArrayProxy = AFNIArrayProxy
+
+    @classmethod
+    @kw_only_meth(1)
+    def from_file_map(klass, file_map, mmap=True):
+        """
+        Parameters
+        ----------
+        file_map : dict
+            dict with keys ``image, header`` and values being fileholder
+            objects for the respective BRIK and HEAD files
+        mmap : {True, False, 'c', 'r'}, optional, keyword only
+            `mmap` controls the use of numpy memory mapping for reading image
+            array data.  If False, do not try numpy ``memmap`` for data array.
+            If one of {'c', 'r'}, try numpy memmap with ``mode=mmap``.  A
+            `mmap` value of True gives the same behavior as ``mmap='c'``.  If
+            image data file cannot be memory-mapped, ignore `mmap` value and
+            read array from file.
+        """
+        with file_map['header'].get_prepare_fileobj('rt') as hdr_fobj:
+            hdr = klass.header_class.from_fileobj(
+                hdr_fobj)
+        brik_fobj = file_map['image'].get_prepare_fileobj()
+        data = klass.ImageArrayProxy(brik_fobj, hdr,
+                                     mmap=mmap)
```

Comment:
I think the issue might be here. I don't think you need to `get_prepare_fileobj` in this method.

```Python
data = klass.ImageArrayProxy(file_map['image'], hdr, mmap=mmap)
```

I think this will help release the filehandles between reads. Please see analyze.py for probably the most exercised (as the Nifti1Image superclass) variant of `from_file_map`.